src/wallet/rpcwallet.cpp

```diff
@@ -4957,8 +4957,16 @@ static RPCHelpMan walletcreatefundedpsbt()
     // Automatically select coins, unless at least one is manually selected. Can
     // be overridden by options.add_inputs.
     coin_control.m_add_inputs = rawTx.vin.size() == 0;
+    // FundTransaction expects blinding keys, if present, to appear in the output nonces
+    for (auto it = rawTx.vout.begin(); it != rawTx.vout.end(); ++it) {
+        auto search_it = psbt_outs.find(*it);
+        assert (search_it != psbt_outs.end());
+        CPubKey& blind_pub = search_it->second.m_blinding_pubkey;
+        if (blind_pub.IsFullyValid()) {
+            it->nNonce.vchCommitment = std::vector<unsigned char>(blind_pub.begin(), blind_pub.end());
+        }
+    }
```

Comment:
In 70faa7e6b7a4967302154c51a1898d173e594f15 "walletcreatefundedpsbt: signal blinding data correctly to `FundTransaction`"

nit: this could be a range based for loop:

```suggestion
    for (CTxOut& txout : rawTx.vout) {
        auto search_it = psbt_outs.find(txout);
        assert (search_it != psbt_outs.end());
        CPubKey& blind_pub = search_it->second.m_blinding_pubkey;
        if (blind_pub.IsFullyValid()) {
            txout.nNonce.vchCommitment = std::vector<unsigned char>(blind_pub.begin(), blind_pub.end());
        }
    }
```